Mathlib/NumberTheory/ModularForms/JacobiTheta/TwoVariable.lean

```diff
@@ -14,119 +15,501 @@ This file defines the two-variable Jacobi theta function
+        replace h := (h.comp_injective this).tendsto_atTop_zero
+        simp_rw [Function.comp_def, Int.cast_neg, Int.cast_ofNat, mul_neg, neg_mul, neg_neg] at h
+        refine atTop_neBot.ne (disjoint_self.mp <| h.disjoint (disjoint_nhds_atTop 0) ?_)
+        exact tendsto_exp_atTop.comp ((tendsto_nat_cast_atTop_atTop.const_mul_atTop
+          (mul_pos two_pos pi_pos)).atTop_mul_const hz)
+
+lemma norm_jacobiTheta₂_term_fderiv_le (n : ℤ) (z τ : ℂ) : ‖jacobiTheta₂_term_fderiv n z τ‖
+    ≤ 3 * π * |n| ^ 2 * ‖jacobiTheta₂_term n z τ‖ := by
+  rw [jacobiTheta₂_term_fderiv, jacobiTheta₂_term, norm_smul (α := ℂ) (β := (ℂ × ℂ) →L[ℂ] ℂ),
+    mul_comm _ ‖cexp _‖, (by norm_num : (3 : ℝ) = 2 + 1), add_mul, add_mul]
+  refine mul_le_mul_of_nonneg_left ((norm_add_le _ _).trans (add_le_add ?_ ?_)) (norm_nonneg _)
+  · simp_rw [norm_smul (α := ℂ) (β := (ℂ × ℂ) →L[ℂ] ℂ), norm_mul, ← ofReal_ofNat, ← ofReal_int_cast,
+      norm_real, norm_of_nonneg zero_le_two, Real.norm_of_nonneg pi_pos.le, norm_I, mul_one,
+      Real.norm_eq_abs, Int.cast_abs, mul_assoc]
+    refine mul_le_mul_of_nonneg_left (mul_le_mul_of_nonneg_left ?_ pi_pos.le) two_pos.le
+    refine le_trans ?_ (?_ : |(n : ℝ)| ≤ |(n : ℝ)| ^ 2)
+    · exact mul_le_of_le_one_right (abs_nonneg _) (ContinuousLinearMap.norm_fst_le ..)
+    · rw [← Int.cast_abs, ← Int.cast_natAbs, ← Nat.cast_pow, Nat.cast_le]
+      apply Nat.le_self_pow two_ne_zero
+  · simp_rw [norm_smul (α := ℂ) (β := (ℂ × ℂ) →L[ℂ] ℂ), norm_mul, one_mul, norm_I, mul_one,
+      norm_real, norm_of_nonneg pi_pos.le, ← ofReal_int_cast, ← ofReal_pow, norm_real,
+      Real.norm_eq_abs, Int.cast_abs, _root_.abs_pow]
+    apply mul_le_of_le_one_right (mul_nonneg pi_pos.le (pow_nonneg (abs_nonneg _) _))
+    exact ContinuousLinearMap.norm_snd_le ..
+
+lemma norm_jacobiTheta₂_term_fderiv_ge (n : ℤ) (z τ : ℂ) :
+    π * |n| ^ 2 * ‖jacobiTheta₂_term n z τ‖ ≤ ‖jacobiTheta₂_term_fderiv n z τ‖ := by
+  have : ‖(jacobiTheta₂_term_fderiv n z τ) (0, 1)‖ ≤ ‖jacobiTheta₂_term_fderiv n z τ‖ := by
+    refine (ContinuousLinearMap.le_opNorm _ _).trans ?_
+    simp_rw [Prod.norm_def, norm_one, norm_zero, max_eq_right zero_le_one, mul_one, le_refl]
+  refine le_trans ?_ this
+  simp_rw [jacobiTheta₂_term_fderiv, jacobiTheta₂_term, ContinuousLinearMap.coe_smul',
+    Pi.smul_apply, ContinuousLinearMap.add_apply, ContinuousLinearMap.coe_smul',
+    ContinuousLinearMap.coe_fst', ContinuousLinearMap.coe_snd', Pi.smul_apply, smul_zero, zero_add,
+    smul_eq_mul, mul_one, mul_comm _ ‖cexp _‖, norm_mul]
+  refine mul_le_mul_of_nonneg_left (le_of_eq ?_) (norm_nonneg _)
+  simp_rw [norm_real, norm_of_nonneg pi_pos.le, norm_I, mul_one,
+    Int.cast_abs, ← abs_intCast, Complex.norm_eq_abs, Complex.abs_pow]
+
+lemma summable_jacobiTheta₂_term_fderiv_iff (z τ : ℂ) :
+    Summable (jacobiTheta₂_term_fderiv · z τ) ↔ 0 < im τ := by
+  constructor
+  · rw [← summable_jacobiTheta₂_term_iff (z := z)]
+    refine fun h ↦ h.norm.of_norm_bounded_eventually _  ?_
```

Comment:
```suggestion
    refine fun h ↦ h.norm.of_norm_bounded_eventually _ ?_
```

Reply:
Yes, this is very annoying. Lean seems to take a ludicrously long time (about 2 seconds) to infer `FiniteDimensional ℝ (ℂ × ℂ →L[ℂ] ℂ)`. I tried adding some extra instances but this seems to make the problem worse, not better.